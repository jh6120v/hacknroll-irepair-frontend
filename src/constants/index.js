import moment from 'moment';

export const FUNC_GO_BACK = 'GO_BACK';
export const FUNC_CLOSE = 'CLOSE';

export const CHAT_MESSAGE = 'CHAT_MESSAGE';

export const FAKE_DATA = [
    {
        id: 'A0003',
        author: '航海王',
        avatar: 'avatar-3.png',
        message: '你好啊小帥哥，連名字都可以這麼威啊～',
        images: null,
        time: +moment()
    },
    {
        id: 'A0004',
        author: '烏魯魯',
        avatar: 'avatar-4.png',
        message: '就是啊！小明，你一定有很多朋友吧！👏👏',
        images: null,
        time: +moment()
    },
    {
        id: 'A0005',
        author: 'John',
        avatar: 'avatar-5.png',
        message: '連我都忍不住想要見你了～😊',
        images: null,
        time: +moment()
    },
    {
        id: 'A0006',
        author: 'Jim',
        avatar: 'avatar-6.png',
        message: '世界很美好❤️',
        images: null,
        time: +moment()
    }
];
