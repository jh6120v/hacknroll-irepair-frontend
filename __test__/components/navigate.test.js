import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { Router } from 'react-router';
import Navigate from '../../src/components/navigate';
import { FUNC_CLOSE, FUNC_GO_BACK } from '../../src/constants';
import { history } from '../../src/store';

describe('test header component.', () => {
    it('test <Navigate /> snapshot.', () => {
        cleanup();

        const { container } = render(<Navigate />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <Navigate /> when set title.', () => {
        cleanup();

        const { getByTestId } = render(<Navigate title="test_title" />);

        expect(getByTestId('display_title').innerHTML).toEqual('test_title');
    });

    it('test <Navigate /> when go back btn exist.', () => {
        cleanup();

        const { getByTestId } = render(<Navigate prev={FUNC_GO_BACK} />);

        expect(getByTestId('display_prev_link').querySelector('a').innerHTML).toEqual('<svg class="sc-bdVaJa fUuvxv" fill="#fff" width="28px" height="28px" viewBox="0 0 1024 1024" rotate="0"><path d="M704 256.8l-64.6-64.8-319.4 320 319.4 320 64.6-64.8-254.6-255.2z"></path></svg>');
    });

    it('test <Navigate /> when go back click once', () => {
        cleanup();

        history.goBack = jest.fn();

        const { getByTestId } = render(<Navigate prev={FUNC_GO_BACK} />);

        fireEvent.click(getByTestId('display_prev_link').querySelector('a'));

        expect(history.goBack).toHaveBeenCalledTimes(1);
    });

    it('test <Navigate /> when setting btn exist.', () => {
        cleanup();

        const { getByTestId } = render(<Router history={history}><Navigate next="FUNC_SETTING" /></Router>);

        expect(getByTestId('display_next_link').innerHTML).toEqual('');
    });
});
