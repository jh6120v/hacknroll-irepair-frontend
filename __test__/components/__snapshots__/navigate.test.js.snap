// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test header component. test <Navigate /> snapshot. 1`] = `
<div
  class="sc-bdVaJa ckITkV"
>
  <div
    class="sc-htpNat egkiwT"
    data-testid="display_prev_link"
  />
  <div
    class="sc-bwzfXH eyLlis"
    data-testid="display_title"
  />
  <div
    class="sc-htpNat sc-bxivhb ubgqH"
    data-testid="display_next_link"
  />
</div>
`;
